mii/deployment.py

```diff
@@ -98,112 +74,39 @@ def deploy(task,
-
-    # add fields for replica deployment
-    replica_pool = _allocate_processes(mii_config.hostfile,
-                                       mii_config.tensor_parallel,
-                                       mii_config.replica_num)
-    replica_configs = []
-    for i, (hostname, gpu_indices) in enumerate(replica_pool):
-        # Reserver port for a LB proxy when replication is enabled
-        port_offset = 1
-        base_port = mii_config.port_number + i * mii_config.tensor_parallel + port_offset
-        tensor_parallel_ports = list(
-            range(base_port,
-                  base_port + mii_config.tensor_parallel))
-        torch_dist_port = mii_config.torch_dist_port + i
-        replica_configs.append(
-            ReplicaConfig(hostname=hostname,
-                          tensor_parallel_ports=tensor_parallel_ports,
-                          torch_dist_port=torch_dist_port,
-                          gpu_indices=gpu_indices))
-    lb_config = LoadBalancerConfig(port=mii_config.port_number,
-                                   replica_configs=replica_configs)
-
-    if deployment_type != DeploymentType.NON_PERSISTENT:
-        create_score_file(deployment_name=deployment_name,
-                          deployment_type=deployment_type,
-                          task=task,
-                          model_name=model,
-                          ds_optimize=enable_deepspeed,
-                          ds_zero=enable_zero,
-                          ds_config=ds_config,
-                          mii_config=mii_config,
-                          model_path=model_path,
-                          lb_config=lb_config)
-
-    if deployment_type == DeploymentType.AML:
-        _deploy_aml(deployment_name=deployment_name, model_name=model, version=version)
-    elif deployment_type == DeploymentType.LOCAL:
-        return _deploy_local(deployment_name, model_path=model_path)
-    elif deployment_type == DeploymentType.NON_PERSISTENT:
-        assert int(os.getenv('WORLD_SIZE', '1')) == mii_config.tensor_parallel, "World Size does not equal number of tensors. When using non-persistent deployment type, please launch with `deepspeed --num_gpus <tensor_parallel>`"
-        provider = MODEL_PROVIDER_MAP[get_provider_name(model, task)]
-        mii.non_persistent_models[deployment_name] = (load_models(
-            get_task_name(task),
-            model,
-            model_path,
-            enable_deepspeed,
-            enable_zero,
-            provider,
-            mii_config),
-                                                      task)
-    else:
-        raise Exception(f"Unknown deployment type: {deployment_type}")
+    if mii_config.deployment_type != DeploymentType.NON_PERSISTENT:
+        create_score_file(mii_config)
+
+    if mii_config.deployment_type == DeploymentType.AML:
+        _deploy_aml(mii_config)
+    elif mii_config.deployment_type == DeploymentType.LOCAL:
+        return _deploy_local(mii_config)
```

Comment:
I think this is a pointless `return` - should probably just call `_deploy_local`